### PR TITLE
Issues/34507 - $_item->getHasChildren() returns true when the pagination is not there for the cart and returns null when the cart is with pagination. 

### LINKED ISSUE #34507: $_item->getChildren() returns empty array for parent cart items when cart is paginated
### Preconditions (*)

1. PHP 7.4
2. Magento 2.4.3-p1 (also replicated in 2.3.5-p1 and 2.4.2)

### Steps to reproduce (*)

1. Override cart item renderer template (Magento_Checkout/cart/item/default.phtml).
2. Add `$_item->getChildren()` to get child items of configurable products - I am using this to get stock levels for child products when a configurable product is added to the cart. **For demo purposes, I have var_dumped `$_item->getHasChildren()` in the screenshots below.**
3. Add enough individual products to the cart to trigger pagination in the cart (by default more than 20 products).
4. Observe that without pagination active, `$_item->getHasChildren()` returns `NULL` for simple products, and `(bool)true` for configurable products. When pagination is triggered, `$item->getHasChildren()` returns `NULL` for all products, including configurables with children.

### Expected result (*)

`var_dump($_item->getHasChildren())` should return `(bool)true` for configurable products. `$_item->getChildren()` should also return an Magento\Quote\Model\Quote\Item\AbstractItem array. **This screenshot shows it working correctly for a cart where pagination has not been triggered.**

### Actual result (*)

`var_dump($_item->getHasChildren())` actually returns `NULL` for configurable products. `$_item->getChildren()` also returns an empty array. **This screenshot is taken once enough products have been added to the cart to trigger pagination.**

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [x] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### PR DESCRIPTION
**Pull request by artskirk**
$_item->getHasChildren() in Magento_Checkout/cart/item/default.phtml returns true when the pagination is not there for the cart and returns null when the cart is with pagination. 

### Description (*)

I have removed addFieldToFilter method before cart pagination. Added an additional method filterCollection method  that
excludes the items that can't be displayed directly.

### Fixed Issues (if relevant)

1. Fixes magento/magento2#34507

### Manual testing scenarios (*)

Steps followed to reproduce issue:

1. Added configurable and non-configurable products to the cart.

2. Print the $_item->getHasChildren() property on the $_items in in Magento_Checkout/cart/item/default.phtml

Observe the function output with pagination and without pagination for the cart.

**Expected Result**
$_item->getHasChildren() returns true when the pagination is not there for the cart and returns true when the cart is with pagination. (See the attached screenshot)

image

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by manavluhar** [comment]
@artskirk kindly sign the Adobe CLA for running automated tests.

**Comment by artskirk** [comment]
> CLA Signed

**Comment by Gerke-W** [comment]
This solution does fix the issue with the $_item->getChilderen() returning null.

However it solution is first colecting both the simple and the configurable, then afterwards it is deleting the item with a parent id. This is done after the pagination has been set as a result the item count is decreased for a single page.

For a full page with one configurable product this does mean there will be 20 products collected, then 1 is removed resulting in only 19 shown items.

In short, this breaks the pagination for the crt.
